core/txpool/legacypool/buffer.go

```diff
@@ -0,0 +1,115 @@
+package legacypool
+
+import (
+	containerList "container/list"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type LRUBuffer struct {
+	capacity int
+	buffer   *containerList.List
+	index    map[common.Hash]*containerList.Element
+	mu       sync.Mutex
+	size     int // Total number of slots used
+}
+
+func NewLRUBuffer(capacity int) *LRUBuffer {
+	return &LRUBuffer{
+		capacity: capacity,
+		buffer:   containerList.New(),
+		index:    make(map[common.Hash]*containerList.Element),
+		size:     0, // Initialize size to 0
+	}
+}
+
+func (lru *LRUBuffer) Add(tx *types.Transaction) {
+	lru.mu.Lock()
+	defer lru.mu.Unlock()
+
+	if elem, ok := lru.index[tx.Hash()]; ok {
+		lru.buffer.MoveToFront(elem)
+		return
+	}
+
+	txSlots := numSlots(tx)
+
+	// Remove elements until there is enough capacity
+	for lru.size+txSlots > lru.capacity && lru.buffer.Len() > 0 {
+		back := lru.buffer.Back()
+		removedTx := back.Value.(*types.Transaction)
+		lru.buffer.Remove(back)
+		delete(lru.index, removedTx.Hash())
+		lru.size -= numSlots(removedTx) // Decrease size by the slots of the removed transaction
+	}
+
+	elem := lru.buffer.PushFront(tx)
```

Comment:
It's strange
according to the rule `fisrt come fist go`
should new tx added to the back, and flush from the front?
in LRUBuffer, flush is from the back, 
but in LRUBufferFastCache flush from the front and call `move item to back` as `movetofront`